src/Polly.Core/Simmy/Behavior/BehaviorChaosStrategy.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Polly.Telemetry;
+
+namespace Polly.Simmy.Behavior;
+
+internal sealed class BehaviorChaosStrategy : MonkeyStrategy
+{
+    private readonly ResilienceStrategyTelemetry _telemetry;
+
+    public BehaviorChaosStrategy(
+        BehaviorStrategyOptions options,
+        ResilienceStrategyTelemetry telemetry)
+        : base(options)
+    {
+        Guard.NotNull(telemetry);
+        Guard.NotNull(options.Behavior);
+
+        _telemetry = telemetry;
+        OnBehaviorInjected = options.OnBehaviorInjected;
+        Behavior = options.Behavior;
+    }
+
+    public Func<OnBehaviorInjectedArguments, ValueTask>? OnBehaviorInjected { get; }
+
+    public Func<ResilienceContext, ValueTask> Behavior { get; }
+
+    protected internal override async ValueTask<Outcome<TResult>> ExecuteCore<TResult, TState>(
+        Func<ResilienceContext, TState, ValueTask<Outcome<TResult>>> callback, ResilienceContext context, TState state)
+    {
+        try
+        {
+            if (await ShouldInject(context).ConfigureAwait(context.ContinueOnCapturedContext))
```

Comment:
nit: invert the condition to exit early and reduce the nesting?

Reply:
The reason I'm not early returning is that in the strategies we "always" end up executing the user's delegate (even in the outcome strategy: when the fault/outcome is null), so by early returning we end up adding unnecessary code, e.g:

```c#
if (!ShouldInject())
{
  return return await ExecuteCallbackSafeAsync();
}

// Inject some chaos
// ...
return return await ExecuteCallbackSafeAsync();

```